packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useCreateFolderPath.ts

```diff
@@ -12,32 +14,29 @@ import {
  * @returns The ID of the final folder in the path.
  */
 export function useCreateFolderPath() {
-  const { synapseClient } = useSynapseContext()
-
   const { mutateAsync: createEntity } = useCreateEntity()
 
+  const queryClient = useQueryClient()
+  const getEntityLookupQueryOptions = useGetEntityLookupQueryOptions()
+  const getEntityQueryOptions = useGetEntityQueryOptions()
+
   return useMutation({
     mutationFn: async (args: { rootContainerId: string; path: string[] }) => {
       const { rootContainerId: rootContainerId, path } = args
       let parentId = rootContainerId
       for (const pathElement of path) {
-        const foundEntityIdResult = await allowNotFoundError(() =>
-          synapseClient.entityServicesClient.postRepoV1EntityChild({
-            entityLookupRequest: {
-              parentId: parentId,
-              entityName: pathElement,
-            },
+        const foundEntityId = await queryClient.fetchQuery(
+          getEntityLookupQueryOptions({
+            parentId: parentId,
+            entityName: pathElement,
           }),
         )
```

Comment:
Fetch with cache to see if entity exists. 

If the entity does not exist, it is created and the cache is invalidated, so a stale cache issue should never happen

Reply:
What would happen in the unlikely event that someone else renames an entity (which is not reflected in our local cache)?  In the case where the new name conflicts, would the process fail during File entity creation?  In the case where the new name no longer conflicts, I suppose the user would be prompted to add a new version for the existing entity?

Reply:
If someone else modifies a node in the tree before we can completely set it up, an error may happen, but I think we have that risk with or without a cache. Ideally we would have an API that would just let us set up the tree in one call.

In your specific scenario, if a folder is renamed before this check, we'll just end up creating a new folder. If a folder is renamed after this check, then we would probably end up uploading the file to the existing folder with the 'wrong' name. 

This set of tasks only sets up folders, and in the context of an upload, if this task fails, an error message is shown and uploads never start. The user can attempt to address the issue (e.g. delete an existing file that has the same name as one of the new folders) or just restart the upload, depending on the error (e.g. a folder existed, but was renamed or deleted when we tried to use it).